core/templates/pages/skill-editor-page/editor-tab/skill-misconceptions-editor/skill-misconceptions-editor.directive.spec.ts

```diff
@@ -0,0 +1,238 @@
+    sampleMisconception = misconceptionObjectFactory.create(
+      'misconceptionId', 'misconceptionName', 'notes', 'feedback', false);
+    sampleSkill._misconceptions = [sampleMisconception];
+
+    spyOn(skillEditorStateService, 'getSkill').and.returnValue(sampleSkill);
+
+    ctrl = $injector.instantiate(directive.controller, {
+      $rootScope: $scope,
+      $scope: $scope,
+      NgbModal: ngbModal,
+      $uibModal: $uibModal,
+      WindowDimensionsService: windowDimensionsService
+    });
+  }));
+
+  beforeEach(() => {
+    testSubscriptions = new Subscription();
+    testSubscriptions.add(skillEditorStateService.onSkillChange.subscribe(
+      skillChangeSpy));
+  });
+
+  afterEach(() => {
+    testSubscriptions.unsubscribe();
+  });
+
+  it('should set properties when initialized', function() {
+    // Misconception list is shown only when window is not narrow.
+    spyOn(windowDimensionsService, 'isWindowNarrow').and.returnValue(false);
+    spyOnProperty(skillEditorStateService, 'onSkillChange').and.returnValue(
+      mockOnSkillChangeEmitter);
+
+    expect($scope.skill).toBe(undefined);
+    expect($scope.misconceptions).toBe(undefined);
+    expect($scope.misconceptionsListIsShown).toBe(undefined);
+
+    ctrl.$onInit();
+    mockOnSkillChangeEmitter.emit();
+
+    expect($scope.skill).toEqual(sampleSkill);
+    expect($scope.misconceptions).toEqual(sampleSkill.getMisconceptions());
+    expect($scope.misconceptionsListIsShown).toEqual(true);
+  });
+
+  it('should toggle misconceptionList', function() {
+    spyOn(windowDimensionsService, 'isWindowNarrow').and.returnValue(true);
+    $scope.misconceptionsListIsShown = false;
+
+    $scope.toggleMisconceptionLists();
+    expect($scope.misconceptionsListIsShown).toBe(true);
+
+    $scope.toggleMisconceptionLists();
+    expect($scope.misconceptionsListIsShown).toBe(false);
+  });
+
+  it('should open add misconception modal when clicking on add ' +
+    'button', function() {
+    let deferred = $q.defer();
+    deferred.resolve({result: sampleMisconception});
+    let modalSpy = spyOn($uibModal, 'open').and.returnValue(
+      {result: deferred.promise});
+    let addMisconceptionSpy = spyOn(
+      skillUpdateService, 'addMisconception').and.returnValue(null);
+
+    $scope.openAddMisconceptionModal();
+    $rootScope.$apply();
+
+    expect(modalSpy).toHaveBeenCalled();
+    expect(addMisconceptionSpy).toHaveBeenCalled();
+  });
+
+  it('should close add misconception modal when clicking on close ' +
+    'button', function() {
+    let deferred = $q.defer();
+    deferred.reject();
+    let modalSpy = spyOn($uibModal, 'open').and.returnValue(
+      {result: deferred.promise});
+    let addMisconceptionSpy = spyOn(
+      skillUpdateService, 'addMisconception').and.returnValue(null);
+
+    $scope.openAddMisconceptionModal();
+    $rootScope.$apply();
+
+    expect(modalSpy).toHaveBeenCalled();
+    expect(addMisconceptionSpy).not.toHaveBeenCalled();
```

Comment:
Is it possible to check what it was called with?

Reply:
done